### PR TITLE
Test that D-Bus works

### PR DESCRIPTION
**Pull request by nievesmontero**
Fixes https://github.com/containers/toolbox/issues/1330

### CODE REVIEW

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:44]: These ShellCheck directives are not useful without running `shellcheck(1)` on the file.  :)

**Review by debarshiray — CHANGES_REQUESTED** [review]
Thanks for working on this!  The new file needs to be added to `test/system/meson.build` so that `shellcheck(1)` gets run on it.

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:34]: The `$(...)` should be enclosed in double quotes.

**Review by debarshiray — CHANGES_REQUESTED** [review]
This only tests the default containers.  ie., Fedora container on a Fedora host, Ubuntu container on a Ubuntu host, etc..  We should test other combinations as well, like we do elsewhere.

Did you look at other things that we might have to check for working D-Bus?

File: test/system/211-dbus.bats
Comment (nievesmontero) [reply test/system/211-dbus.bats:44]: Added :)

File: test/system/211-dbus.bats
Comment (nievesmontero) [reply test/system/211-dbus.bats:34]: Done.

**Comment by nievesmontero** [comment]
> This only tests the default containers. ie., Fedora container on a Fedora host, Ubuntu container on a Ubuntu host, etc.. We should test other combinations as well, like we do elsewhere.
> 
> Did you look at other things that we might have to check for working D-Bus?

I think we should maybe also test on different RHEL versions, like we do in most tests.

**Comment by nievesmontero** [comment]
recheck

**Comment by nievesmontero** [comment]
recheck

**Comment by nievesmontero** [comment]
recheck

File: test/system/104-run.bats
Comment (debarshiray) [inline test/system/104-run.bats:247]: I see that you have commented out this test, but in the test logs it still seems to be running:
```
fedora-38 | not ok 127 run: Ensure that /home/zuul-worker is used as a fallback working directory in 3054ms
fedora-38 | # (in test file test/system/104-run.bats, line 251)
fedora-38 | #   `run -127 --separate-stderr $TOOLBOX run pwd' failed, expected exit code 127, got 0
fedora-38 | # /etc/kernel ~/src/github.com/containers/toolbox
fedora-38 | # /home/zuul-worker
fedora-38 | # stderr:
fedora-38 | # Error: crun: chdir `/etc/kernel`: No such file or directory: OCI runtime attempted to invoke a command that was not found
fedora-38 | # Error: directory /etc/kernel not found in container fedora-toolbox-38
fedora-38 | # Using /home/zuul-worker instead.
```

I am not sure what's going on.  :)

**Comment by debarshiray** [comment]
> > This only tests the default containers. ie., Fedora container on a Fedora host, Ubuntu container on a Ubuntu host, etc.. We should test other combinations as well, like we do elsewhere.
> > Did you look at other things that we might have to check for working D-Bus?
> 
> I think we should maybe also test on different RHEL versions, like we do in most tests.

Yes, that's what I meant by *test other combinations*.  :)

**Comment by debarshiray** [comment]
Note that there's an ongoing Zuul CI breakage with Fedora Rawhide hosts.  See the failures in https://github.com/containers/toolbox/pull/1344 and https://github.com/containers/toolbox/pull/1331  So far this looks like a `rsync` [bug](https://bugzilla.redhat.com/show_bug.cgi?id=2229654) in Fedora Rawhide.

However the tests should still pass on the other hosts.  ie., CentOS Stream 9, Fedoras 37 and 38, and Ubuntu.

File: test/system/104-run.bats
Comment (debarshiray) [reply test/system/104-run.bats]: Okay, it dropped out of the test logs after the last `recheck`.  Now let's see why it was failing.

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:42]: The line count assertion needs to be conditionalized for Bats >= 1.10.0.  See [this](https://github.com/containers/toolbox/commit/f716b23914b2429d1284b7c0c68e9d14f29bbd73) commit for an example.

Fedora 39 has Bats 1.10.0 which fixed a [bug](https://github.com/bats-core/bats-core/issues/708) with the line counting, but other versions of Fedora still have an older Bats.  The line count is actually supposed to be 1, but we were testing for 2 because of the Bats bug.

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:58]: Also, here.

**Review by debarshiray — CHANGES_REQUESTED** [review]
Because of the ongoing Zuul CI [breakage](f716b23914b2429d1284b7c0c68e9d14f29bbd73) with Fedora Rawhide, I ran the tests locally on a Fedora 39 virtual machine.

**Comment by debarshiray** [comment]
> Because of the ongoing Zuul CI [breakage](f716b23914b2429d1284b7c0c68e9d14f29bbd73) with Fedora Rawhide, I ran the tests locally on a Fedora 39 virtual machine.

I used [this](https://kojipkgs.fedoraproject.org/compose/branched/Fedora-39-20230815.n.0/compose/Workstation/x86_64/iso/Fedora-Workstation-Live-x86_64-39-20230815.n.0.iso) Fedora 39 Workstation nightly ISO to create the VM in GNOME Boxes.  You can try it if you want to.

File: test/system/104-run.bats
Comment (debarshiray) [reply test/system/104-run.bats]: I took the liberty to `git push --force` to your branch to restore the tests and the logs say that it's passing:
```
fedora-38 | ok 127 run: Ensure that /home/zuul-worker is used as a fallback working directory in 3208ms
```

Are you sure you didn't have any other local changes?

If you want, you can re-create your local branch with:
```
$ git checkout main
$ git branch -D dbus-test
$ git fetch -p nievesmontero
$ git branch --track dbus-test nievesmontero/dbus-test
```

Comment (nievesmontero) [reply test/system/104-run.bats]: I've followed the steps and re-created my local branch.

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:34]: Nitpick: it will be good to wrap the lines are too long (ie., more than 120 characters).

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:55]: Same here.

**Review by debarshiray — COMMENTED** [review]
Thanks for the update, @nievesmontero !

File: test/system/211-dbus.bats
Comment (debarshiray) [reply test/system/211-dbus.bats]: Fixed.  I took the liberty to update the branch.

File: test/system/211-dbus.bats
Comment (debarshiray) [reply test/system/211-dbus.bats]: Done.

**Comment by debarshiray** [comment]
> > > This only tests the default containers. ie., Fedora container on a Fedora host, Ubuntu container on a Ubuntu host, etc.. We should test other combinations as well, like we do elsewhere.
> > > Did you look at other things that we might have to check for working D-Bus?
> > 
> > 
> > I think we should maybe also test on different RHEL versions, like we do in most tests.
> 
> Yes, that's what I meant by _test other combinations_. :)

@nievesmontero I will merge this once the CI finishes.  Please open a new pull request for the tests combining different host and containers.

File: test/system/211-dbus.bats
Comment (debarshiray) [inline test/system/211-dbus.bats:35]: You need to use a backslash (ie., `\`) when wrapping lines.

Currently the tests are failing with:
```
fedora-38 | not ok 264 dbus: session bus in 180ms
fedora-38 | # (in test file test/system/211-dbus.bats, line 35)
fedora-38 | #   `--object-path /org/freedesktop/DBus --method org.freedesktop.DBus.Peer.Ping)"' failed with status 127
fedora-38 | # Error: Object path is not specified
fedora-38 | # /home/zuul-worker/src/github.com/containers/toolbox/test/system/211-dbus.bats: line 36: --object-path: command not found
fedora-38 | not ok 265 dbus: system bus in 183ms
fedora-38 | # (in test file test/system/211-dbus.bats, line 60)
fedora-38 | #   `org.freedesktop.systemd1.Manager Version)"' failed with status 127
fedora-38 | # Error: Object path is not specified
fedora-38 | # /home/zuul-worker/src/github.com/containers/toolbox/test/system/211-dbus.bats: line 61: --object-path: command not found
fedora-38 | # /home/zuul-worker/src/github.com/containers/toolbox/test/system/211-dbus.bats: line 62: org.freedesktop.systemd1.Manager: command not found
```

**Review by debarshiray — CHANGES_REQUESTED** [review]
Looks like the branch got overwritten before I could merge it.  Like I said before, I had already fixed the line wrapping to save some time.

Now the tests are failing:

File: test/system/211-dbus.bats
Comment (debarshiray) [reply test/system/211-dbus.bats]: Fixed.

File: test/system/211-dbus.bats
Comment (debarshiray) [reply test/system/211-dbus.bats:42]: I think I introduced this mistake in https://github.com/containers/toolbox/pull/1355

**Pull request merged**
